src/group_theory/perm/support.lean

```diff
@@ -177,10 +177,16 @@ lemma is_swap.of_subtype_is_swap {p : α → Prop} [decidable_pred p]
 let ⟨⟨x, hx⟩, ⟨y, hy⟩, hxy⟩ := h in
 ⟨x, y, by { simp only [ne.def] at hxy, exact hxy.1 },
   equiv.ext $ λ z, begin
-    rw [hxy.2, of_subtype],
-    simp only [swap_apply_def, coe_fn_mk, swap_inv, subtype.mk_eq_mk, monoid_hom.coe_mk],
-    split_ifs;
-    rw subtype.coe_mk <|> cc,
+    rw [hxy.2, swap_apply_def],
+    split_ifs,
```

Comment:
Please name the variables explicitly using `split_ifs with h1 h2` etc...
The auto-generated `h_1` and `h_2` that you refer to below are not very stable.

What is the reason that this proof became longer? Are we missing a simp-lemma for the new definition above?

Reply:
I'll do the renaming. 

The reason the proof becomes longer is, I believe, the same for which I need a few lines to prove that extend_domain and of_subtype gave equivalent definitions. I will think about it.